engines/cryomni3d/versailles/engine.cpp

```diff
@@ -0,0 +1,1594 @@
+
+	_dialogsMan.loadGTO("DIALOG1.GTO");
+	setupDialogVariables();
+	setupDialogShows();
+
+	setupPaintingsTitles();
+	setupImgScripts();
+
+	_mainPalette = new byte[3 * 256];
+	setupFonts();
+	setupSprites();
+	loadCursorsPalette();
+
+	// Objects need messages and sprites
+	setupObjects();
+
+	_transparentPaletteMap = new byte[256];
+	_transparentSrcStart = 0;
+	_transparentSrcStop = 240;
+	_transparentDstStart = 0;
+	_transparentDstStop = 248;
+	_transparentNewStart = 248;
+	_transparentNewStop = 254;
+
+	// Inventory has a size of 50
+	_inventory.init(50, new Common::Functor1Mem<unsigned int, void, Toolbar>(&_toolbar,
+	                &Toolbar::inventoryChanged));
+
+	// Init toolbar after we have setup sprites and fonts
+	_toolbar.init(&_sprites, &_fontManager, &_messages, &_inventory, this);
+
+	_fixedImage = new ZonFixedImage(*this, _inventory, _sprites, &kFixedImageConfiguration);
+
+	// Documentation is needed by noone at init time, let's do it last
+	initDocPeopleRecord();
+	_docManager.init(&_sprites, &_fontManager, &_messages, this);
+
+	_countdownSurface.create(40, 15, Graphics::PixelFormat::createFormatCLUT8());
+
+	initGraphics(640, 480);
+	setMousePos(Common::Point(320, 200));
+
+	syncSoundSettings();
+
+	_isPlaying = false;
+	_isVisiting = false;
+
+#if !defined(DEBUG_FAST_START) || DEBUG_FAST_START<1
+	playTransitionEndLevel(-2);
+	if (g_engine->shouldQuit()) {
+		return Common::kNoError;
+	}
+	playTransitionEndLevel(-1);
+	if (g_engine->shouldQuit()) {
+		return Common::kNoError;
+	}
+#endif
+
+	bool stopGame = false;
+	while (!stopGame) {
+		bool exitLoop = false;
+		unsigned int nextStep = 0;
+#if defined(DEBUG_FAST_START) && DEBUG_FAST_START>=2
+		nextStep = 27;
+		// Called in options
+		syncOmni3DSettings();
+#endif
+		setCursor(181);
+		while (!exitLoop) {
+			_isPlaying = false;
+			if (!nextStep) {
+				nextStep = displayOptions();
+			}
+			if (nextStep == 40) {
+				// Quit action
+				exitLoop = true;
+				stopGame = true;
+			} else if (nextStep == 27 || nextStep == 28 || nextStep == 65) {
+				// New game, Load game, Next level
+				if (nextStep == 27) {
+					// New game
+#if !defined(DEBUG_FAST_START) || DEBUG_FAST_START<1
+					playTransitionEndLevel(0);
+					if (g_engine->shouldQuit()) {
+						stopGame = true;
+						exitLoop = true;
+						break;
+					}
+#endif
+					changeLevel(1);
+				} else if (nextStep == 28) {
+					// Load game
+					loadGame(_isVisiting, _loadedSave);
+				} else if (nextStep == 65) {
+					changeLevel(_currentLevel + 1);
+				}
+
+				_isPlaying = true;
+				_toolbar.setInventoryEnabled(!_isVisiting);
+				nextStep = 0;
+				_abortCommand = AbortNoAbort;
+
+				gameStep();
+
+				// We shouldn't return from gameStep without an abort command
+				assert(_abortCommand != AbortNoAbort);
+				switch (_abortCommand) {
+				default:
+					error("Invalid abortCommand: %d", _abortCommand);
+					// Shouldn't return
+					return Common::kUnknownError;
+				case AbortFinished:
```

Comment:
please consider renaming them to kAbort... in accordance with our code naming conventions.